packages/manager/cypress/e2e/core/linodes/create-linode.spec.ts

```diff
@@ -30,12 +29,95 @@ import {
   mockGetLinodeTypes,
 } from 'support/intercepts/linodes';
 
+import type { Region } from '@linode/api-v4';
+
+const mockRegions: Region[] = [
+  regionFactory.build({
+    country: 'uk',
+    id: 'eu-west',
+    label: 'London, UK',
+  }),
+  regionFactory.build({
+    country: 'sg',
+    id: 'ap-south',
+    label: 'Singapore, SG',
+  }),
+  regionFactory.build({
+    id: 'us-east',
+    label: 'Newark, NJ',
+  }),
+  regionFactory.build({
+    id: 'us-central',
+    label: 'Dallas, TX',
+  }),
+];
+
 authenticate();
 describe('create linode', () => {
   before(() => {
     cleanUp('linodes');
   });
 
+  /*
+   * Region select test.
+   * - Confirms that region select dropdown is visible and interactive.
+   * - Confirms that region select dropdown is populated with expected regions.
+   * - Confirms that region select dropdown is sorted alphabetically by region, with North America first.
+   * - Confirms that region select dropdown is populated with expected DCs, sorted alphabetically.
+   * - Confirms that region select dropdown is populated with expected fake DC.
+   */
+  it('region select', () => {
+    mockGetRegions(mockRegions).as('getRegions');
+
+    mockAppendFeatureFlags({
+      soldOutTokyo: makeFeatureFlagData(true),
+    }).as('getFeatureFlags');
+    mockGetFeatureFlagClientstream().as('getClientStream');
+
+    cy.visitWithLogin('linodes/create');
+
+    cy.wait(['@getFeatureFlags', '@getRegions']);
+
+    // Confirm that region select dropdown is visible and interactive.
+    const autocomplete = cy.findByTestId('region-select');
+    ui.regionSelect.find().click();
+
+    cy.get('[data-qa-autocomplete-popper="true"]').should('be.visible');
+
+    // Confirm that region select dropdown are grouped by region,
+    // sorted alphabetically, with North America first.
+    cy.get('[data-qa-region-select-group]')
+      .should('have.length', 3)
+      .should((group) => {
+        expect(group[0]).to.contain('North America');
+        expect(group[1]).to.contain('Asia');
+        expect(group[2]).to.contain('Europe');
+      });
+
+    // Confirm that region select dropdown is populated with expected regions, sorted alphabetically.
+    cy.get('[data-qa-option]')
+      .should('have.length', 5)
+      .should((item) => {
+        expect(item[0]).to.contain('Dallas, TX (us-central)');
+        expect(item[1]).to.contain('Newark, NJ (us-east)');
+        expect(item[2]).to.contain('Singapore, SG (ap-south)');
+        expect(item[3]).to.contain('Tokyo, JP (ap-northeast) (Not available)');
+        expect(item[4]).to.contain('London, UK (eu-west)');
+      });
+
+    cy.get('[data-qa-option="eu-west"]').click();
+    cy.get('[data-qa-autocomplete-popper="true"]').should('not.exist');
+    autocomplete.within(() => {
+      cy.get('[data-testid="textfield-input"]').should(
+        'have.value',
+        'London, UK (eu-west)'
+      );
+    });
```

Comment:
```suggestion
    ui.regionSelect.find().within(() => {
      cy.get('[data-testid="textfield-input"]').should(
        'have.value',
        'London, UK (eu-west)'
      );
    });
// (And line 82 can be removed.)
```
This is an [antipattern](https://docs.cypress.io/guides/references/best-practices#Assigning-Return-Values) in Cypress and can cause unexpected test behavior.

If you want to refer to an element that was selected before, you'll want to do it [with an alias](https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Sharing-Context) instead of a const assignment, for example:

```typescript
ui.regionSelect.find().as('autocomplete');

// Later...
cy.get('@autocomplete').within(() => { ... })
```

(I think, essentially, if the page or component re-renders after you assign the const, it'll fail when you try to use it later in the test)